website/blog/2018-06-25-on-consuming-and-publishing-es6+-packages.md

```diff
@@ -0,0 +1,276 @@
+
+Babel used to be [`6to5`](https://babeljs.io/blog/2017/10/05/babel-turns-three), since it only converted from ES2015 (known as ES6 back then) to ES5. Back then, the browser support for ES2015 was almost non-existent, so the idea of a JavaScript compiler was both novel and useful: we could write modern code, and have it work for all of our users.
+
+But what about the browser runtimes themselves? Because evergreen browsers will eventually catch up to the standard (as they have with ES2015), creating `preset-env` helps Babel and the community align with both the browsers and TC39 itself. If we only compiled to ES5, no one would ever run native code in the browsers.
+
+The real difference is realizing that there will _always_ be a sliding window of support:
+
+- Application code (our supported environments)
+- Browsers (Chrome, Firefox, Edge, Safari)
+- Babel (the abstraction layer)
+- TC39/ECMAScript proposals (and Babel implementations)
+
+Thus, the need isn't just for `6to5` to be renamed to Babel because it compiles to `7to5`, but for Babel to change the implicit assumption it only targets ES5. With `@babel/preset-env`, we are able to write the latest JavaScript and target whichever browser/environment!
+
+Using Babel and `preset-env` helps us keep up with that constantly changing sliding window. However, even if we use it, it's currently used only for *our application code*, and not for our code’s dependencies.
+
+## Who Owns Our Dependencies?
+
+Because we have control over our own code, we are able to take advantage of `preset-env`: both by writing in ES2015+ and targeting ES2015+ browsers.
+
+This isn't necessarily the case for our dependencies; in order to get the same benefits as compiling our code we may need to make some changes.
+
+Is it as straightforward as just running Babel over `node_modules`?
+
+## The Current Complexities in Compiling Dependencies
+
+### Compiler Complexity
+
+Although it shouldn't deter us from making this a common practice, we should be aware that compiling dependencies does increase the surface area of issues and complexity.
+
+- Compilers are no different than other programs and have bugs. 
+- `preset-env` itself could have bugs because we use [`compat-table`](https://kangax.github.io/compat-table/es6/) for our data vs. [Test262](https://github.com/tc39/test262) (the official test suite).
+- Browsers themselves can have issues with running native ES2015+ code vs. ES5.
+- There is still a question of determining what is "supported": https://github.com/babel/babel-preset-env/issues/54 is an example of an edge case.
+
+#### Specific Issues in v6
+
+Running a `script` as a `module` either causes a `SyntaxError`, new runtime errors, or unexpected behavior due to the differences in semantics between classic scripts and modules.
```

Comment:
We enumerate these differences in detail here: https://developers.google.com/web/fundamentals/primers/modules#intro Maybe worth a link?